actionpack/lib/action_dispatch/testing/integration.rb

```diff
@@ -239,6 +239,7 @@ def process(method, path, params: nil, headers: nil, env: nil, xhr: false, as: n
         end
 
         hostname, port = host.split(":")
+        input = params.delete(:input) if params.is_a?(Hash)
```

Comment:
I think the change makes sense, but I'm not sure about it being nested under `params` if we're going to remove it anyway. Is there a reason to not make it another optional keyword argument?

Since this is a new feature you should also add a changelog entry

Reply:
Thanks, I like the optional keyword argument better.